nursery/coin_select/src/metrics/lowest_fee.rs

```diff
@@ -0,0 +1,175 @@
+use crate::{
+    float::Ordf32, metrics::change_lower_bound, BnbMetric, Candidate, CoinSelector, Drain,
+    DrainWeights, FeeRate, Target,
+};
+
+/// Metric that aims to minimize transaction fees. The future fee for spending the change output is
+/// included in this calculation.
+///
+/// The scoring function for changeless solutions is:
+/// > input_weight * feerate + excess
+///
+/// The scoring function for solutions with change:
+/// > (input_weight + change_output_weight) * feerate + change_spend_weight * long_term_feerate
+pub struct LowestFee<'c, C> {
+    /// The target parameters for the resultant selection.
+    pub target: Target,
+    /// The estimated feerate needed to spend our change output later.
+    pub long_term_feerate: FeeRate,
+    /// Policy to determine the change output (if any) of a given selection.
+    pub change_policy: &'c C,
+}
+
+impl<'c, C> Clone for LowestFee<'c, C> {
+    fn clone(&self) -> Self {
+        *self
+    }
+}
+
+impl<'c, C> Copy for LowestFee<'c, C> {}
+
+impl<'c, C> LowestFee<'c, C>
+where
+    for<'a, 'b> C: Fn(&'b CoinSelector<'a>, Target) -> Drain,
+{
+    fn calc_metric(&self, cs: &CoinSelector<'_>, drain_weights: Option<DrainWeights>) -> f32 {
+        self.calc_metric_lb(cs, drain_weights)
+            + match drain_weights {
+                Some(_) => {
+                    let selected_value = cs.selected_value();
+                    assert!(selected_value >= self.target.value);
+                    (cs.selected_value() - self.target.value) as f32
```

Comment:
what's going on here? When you do have a change output do you add this extra input minus target to the value :confused:.

Reply:
Should this rather be in the `None` case for the excess? I've just encountered one scenario where I get the same score for two different changeless selections, but one of these selections has greater excess.